pandas/types/concat.py

```diff
@@ -201,6 +201,43 @@ def convert_categorical(x):
         return Categorical(concatted, rawcats)
 
 
+def union_categoricals(to_union):
+    """
+    Combine list-like of Categoricals, unioning categories. All
+    must have the same dtype, and none can be ordered.
+
+    Parameters
+    ----------
+    to_union : list like of Categorical
+
+    Returns
+    -------
+    Categorical
+       A single array, categories will be ordered as they
+       appear in the list
+    """
+    from pandas import Index, Categorical, unique
+
+    if any(c.ordered for c in to_union):
+        raise TypeError("Can only combine unordered Categoricals")
+
+    first = to_union[0]
+    if not all(com.is_dtype_equal(c.categories, first.categories)
+               for c in to_union):
+        raise TypeError("dtype of categories must be the same")
+
+    unique_cats = unique(np.concatenate([c.categories for c in to_union]))
```

Comment:
Is this really safe if `categories` is an index dtype with a pandas-specific dtype? e.g., datetime64 with timezone or period.

That's why I thought we needed to use the Index.append() method.